File: String/Count Binary Substrings.py
class Solution:
    def countBinarySubstrings(self, s):
        """
        :type s: str
        :rtype: int
        """
        chunks = []
        consective = 1
        for i in range(1, len(s)):
            if s[i] == s[i-1]:
                consective += 1
            else:
                chunks.append(consective)
                consective = 1
        chunks.append(consective)
        pairs = zip(chunks[1:], chunks[:-1])
        return sum([min(i) for i in pairs])

File: String/Count Binary Substrings.cpp
class Solution {
public:
    int countBinarySubstrings(string s) {
        int count = 0;
        int consective = 1;
        vector<int> chunks;
        for(int i=1; i<s.size(); i++){
            if(s[i] == s[i-1]){
                consective += 1;
            }
            else{
                chunks.push_back(consective);
                consective = 1;
            }
        }
        chunks.push_back(consective);
        for(int j=0; j<chunks.size()-1; j++){
            count += min(chunks[j], chunks[j+1]);
        }
        return count;
    }
};